test/integration/ping-pong.test.js

```diff
@@ -0,0 +1,136 @@
+
+        const volumeLogger = new VolumeLogger(0)
+        websocketServer = new WebsocketServer(
+            uWS.App(),
+            wsPort,
+            networkNode,
+            new StreamFetcher('http://localhost:8081/streamr-core'),
+            new Publisher(networkNode, volumeLogger),
+            volumeLogger,
+            new SubscriptionManager(networkNode)
+        )
+
+        client1 = createClient(wsPort, 'tester1-api-key')
+        await client1.ensureConnected()
+
+        client2 = createClient(wsPort, 'tester1-api-key')
+        await client2.ensureConnected()
+
+        client3 = createClient(wsPort, 'tester1-api-key')
+        await client3.ensureConnected()
+
+        await waitForCondition(() => websocketServer.connections.size === 3)
+    })
+
+    afterEach(async () => {
+        await client1.ensureDisconnected()
+        await client2.ensureDisconnected()
+        await client3.ensureDisconnected()
+
+        await tracker.stop()
+        await networkNode.stop()
+        await websocketServer.close()
+    })
+
+    it('websocketServer sends pings and receives pongs from clients', async () => {
+        let pings = 0
+
+        const connections = [...websocketServer.connections.values()]
+        connections.forEach((connection) => {
+            expect(connection.isAlive).toBeUndefined()
+        })
+
+        client1.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        client2.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        client3.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        // eslint-disable-next-line no-underscore-dangle
+        websocketServer._pingConnections()
+        await waitForCondition(() => pings === 3)
+
+        expect(pings).toEqual(3)
+
+        connections.forEach((connection) => {
+            expect(connection.isAlive).toBeTruthy()
+        })
+    })
+
+    it('websocketServer closes connections, which are not replying with pong', async (done) => {
+        let pings = 0
+
+        client1.connection.socket.pong = () => {
+            // don't send back pong
+        }
+
+        client2.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        client3.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        // eslint-disable-next-line no-underscore-dangle
+        websocketServer._pingConnections()
+        await waitForCondition(() => pings === 2)
+
+        const connections = [...websocketServer.connections.values()]
+        connections.forEach((connection, index) => {
+            // first client
+            if (index === 0) {
+                expect(connection.isAlive)
+                    .toBeFalsy()
+            } else {
+                expect(connection.isAlive)
+                    .toBeTruthy()
+            }
+        })
+
+        client1.on('disconnected', () => {
```

Comment:
I don't understand why this and then `websocketServer._pingConnections()` at end of this test is needed.

Reply:
- on first call of `_pingConnections` we set `respondedPong=false` for each websocket
- then we expect to get `pong` frame and set `respondedPong=true`
- after `pingInterval` seconds, we call again `_pingConnections`
- if `respondedPong=false`, it means that websocket didn't send back `pong` frame -> disconnect not active `websocket`
- if `respondedPong=true`  -> we got `pong` frame, everything is good, repeat from first step